app/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsService.scala

```diff
@@ -109,24 +113,85 @@ class DefaultReturnsService @Inject() (
         returnRequest.subscribedDetails.cgtReference,
         desSubmitReturnRequest
       )
-      .subflatMap { response =>
+      .map { httpResponse =>
```

Comment:
there's no need to have an additional `map` here - you can close the timer and audit the response before you pattern match inside the `subflatMap`.

Reply:
Done.

Reply:
Done.